src/components/keyboard-nav.js

```diff
@@ -0,0 +1,93 @@
+import * as _ from 'lodash/fp'
+import { useRef } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { toIndexPairs } from 'src/libs/utils'
+
+
+export const withArrowKeyNavigation = ({ onKeyDown, children }) => {
+  const count = children.length
+  const refs = _.map(() => useRef(), _.range(0, count))
+
+  const focusOn = index => {
+    // Wrap around the ends and ensure the number is positive
+    const i = index % count + (index < 0 ? count : 0)
+    refs[i].current && refs[i].current.focus()
+  }
+
+  // Pass a ref into each child, using an alternative property name
+  return _.map(([i, child]) => {
+    return h(child, { key: i, forwardedRef: refs[i], onKeyDown: onKeyDown(i, focusOn) })
```

Comment:
This keying by index is likely to cause issues if the children are re-ordered or the number of children changes. Can we use something more durable?

Reply:
Would it be legit to see if the child itself has a key property and if so use that, using the index as a fallback?

Reply:
No, I guess that wouldn't work since these children are functions which haven't been rendered yet.

2 potential options:

1. Don't use `h` to create wrapper elements at all, and just call the chlidren directly as functions. You'd lose the convenient type coersion, but it's possible the key property wouldn't be necessary anymore.

2. Pass in a keyed object of children rather than an array. This would force the caller to create keys for all their elements, which may or may not make much sense in context. For tabs it would be fine since they already have keys, but for menu items it may not. It wouldn't solve the `useRef` issue if the number of children changes, however.

Are there any other options?